### ISSUE TITLE
Classes, Jobs, DoL, DoH, Eureka and Bozja levels are always null

### ISSUE THREAD

**Issue by tancred423** [xivapi/nodestone#7]
# Situation
The result of a character request by ID always shows classes, etc. as null. I found the issue and made a fix, but I rather not make a pull request as this is a very dirty fix. I will still explain what I did and what caused the issue in case you want to fix it.

This is the result currently:
![image](https://user-images.githubusercontent.com/38757664/143726864-485ec5b2-e8a7-4e52-b5fb-c6384c99eb86.png)

# Explaining the problem
In the [page-parser.ts](https://github.com/xivapi/nodestone/blob/master/src/core/page-parser.ts) only the DOM from the character profile will be requested, so `/lodestone/character/:id`. But not from the Class/Job overview, so `/lodestone/character/:id/class_job`. That is the one we need in order to apply the css query selectors for the classes.

# My (dirty) fix for that
What i just did now is, I added another DOM request at the start of the `parse()` function that is very similar to the existing one, but for the `/class_job` path:
```js
const classJobDataAll = await axios.get(`${this.getURL(req)}/class_job`).catch((err: any) => {
   throw new Error(err.response.status)
})

const classJobData = classJobDataAll.data
const classJobDom = new JSDOM(classJobData)
let classJobDocument = classJobDom.window.document
```

Now we have `document` for all the general info and `classJobDocument` for the classes.

Then later, when we call `this.handleColumn()` in currently line 54, we have to pass in the `classJobDocument` instead of `document` if the column is one of the classes.

For that _(and this is the very dirty part)_, I just made a simple array with all classes and checked the column:
```js
let correctDocument = document
if (
   ['Bozja', 'Eureka', 'Paladin', 'Warrior', 'Darkknight', 'Gunbreaker', 'Monk', 'Dragoon', 'Ninja', 'Samurai', 'Whitemage', 'Scholar', 'Astrologian', 'Bard', 'Machinist', 'Dancer', 'Blackmage', 'Summoner', 'Redmage', 'Bluemage', 'Carpenter', 'Blacksmith', 'Armorer', 'Goldsmith', 'Leatherworker', 'Weaver', 'Alchemist', 'Culinarian', 'Miner', 'Botanist', 'Fisher']
      .includes(column)
) {
   correctDocument = classJobDocument
} else correctDocument = document
```
Now we just have to pass `correctDocument` to `this.handleColumn()` instead of `document` and it will work again.

And this is the result with the same character after the fix:
![image](https://user-images.githubusercontent.com/38757664/143726895-7faa77f6-66c0-406d-bc86-abcaa2e6dbfd.png)
(Eureka is in fact null, because I never played that, but I saw the Eureka info on a friends char!)

**Comment by Supamiu** [comment]
This looks like a duplicate of https://github.com/xivapi/lodestone-css-selectors/issues/12

**Comment by nazdir** [comment]
I don't think this is a duplicate of the other issue.  Parsing with Nodestone returns `null` for all classes.  The other issue was only related to Eureka/Bozja.
